webapp/app/routes/app/freeze_.$id/contract-write-button.tsx

```diff
@@ -0,0 +1,82 @@
+import type { signaturesTable } from "@/.server/db/schema";
+import { Button } from "@/components/ui/button";
+import { dateToUnixTimestamp } from "@/utils/date";
+import { ALL_ABIS } from "@/utils/raw-abis";
+import { useNavigate } from "@remix-run/react";
+import type { InferSelectModel } from "drizzle-orm";
+import type React from "react";
+import { toast } from "react-hot-toast";
+import { $path } from "remix-routes";
+import type { ContractFunctionName, Hex } from "viem";
+import { useAccount, useWriteContract } from "wagmi";
+
+type BroadcastTxButtonProps = {
+  target: Hex;
+  signatures: InferSelectModel<typeof signaturesTable>[];
+  threshold: number;
+  disabled?: boolean;
+  children?: React.ReactNode;
+  validUntil: Date;
+  functionName: ContractFunctionName<typeof ALL_ABIS.council, "nonpayable">;
+};
+
+export default function ContractWriteButton({
```

Comment:
I feel that in the future we can unify all the contract write buttons doing a little bit of magic. Not on this PR, but I'll open a low priority task for this.